### PR TITLE
add support and probe builtin clusters

### PR DESCRIPTION
**Pull request by jubrad**
- Adds mz_probe cluster to be use by mz_monitoring user to collect uptime data.

- Adds mz_support cluster to be used by support users. This will be owned by support and will not have any replicas to start with. The default size match the mz_system cluster size for now.

### Motivation

  * This PR adds a known-desirable feature.
https://github.com/MaterializeInc/materialize/issues/25834

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by benesch** [comment]
Woohoo! So excited for this. Can you also add the new builtin clusters to the docs in doc/user/content/sql/show-clusters.md? You can use the `{{< warn-if-unreleased >}}` tag to version-gate the content.

File: src/adapter/src/catalog/open.rs [hunk 1]
```diff
@@ -1803,6 +1806,10 @@ impl BuiltinBootstrapClusterSizes {
             Ok(self.system_cluster.clone())
         } else if cluster_name == mz_catalog::builtin::MZ_INTROSPECTION_CLUSTER.name {
             Ok(self.introspection_cluster.clone())
+        } else if cluster_name == mz_catalog::builtin::MZ_PROBE_CLUSTER.name {
+            Ok(self.probe_cluster.clone())
+        } else if cluster_name == mz_catalog::builtin::MZ_SUPPORT_CLUSTER.name {
+            Ok(self.system_cluster.clone())
```
Comment (jubrad) [inline src/adapter/src/catalog/open.rs:1812]: I'm not so sure about defaulting this to mz_system.

Comment (benesch) [reply src/adapter/src/catalog/open.rs:1808]: Is it time to bite the bullet and introduce separate, dedicated CLI args for each of the builtin cluster's sizes?

File: test/testdrive/catalog.td [hunk 2]
```diff
@@ -419,8 +419,7 @@ $ postgres-execute connection=postgres://mz_system@${testdrive.materialize-inter
 ALTER SYSTEM SET enable_rbac_checks TO true
 
 # Schema names that start with "mz_" or "pg_" are reserved for future use by the
-# system.
-! CREATE SCHEMA mz_foo
+# system. ! CREATE SCHEMA mz_foo
```
Comment (benesch) [inline test/testdrive/catalog.td:422]: This looks like an accidental but load bearing rewrap!

Comment (jubrad) [reply test/testdrive/catalog.td:422]: eeek! nice catch.

File: src/catalog/src/builtin.rs [hunk 3]
```diff
@@ -6725,6 +6728,42 @@ pub const MZ_INTROSPECTION_CLUSTER_REPLICA: BuiltinClusterReplica = BuiltinClust
...
+        },
+        MzAclItem {
+            grantee: MZ_MONITOR_ROLE_ID,
+            grantor: MZ_SYSTEM_ROLE_ID,
+            acl_mode: AclMode::USAGE,
+        },
+        rbac::owner_privilege(ObjectType::Cluster, MZ_SYSTEM_ROLE_ID),
+    ],
+};
+// maybe we can get rid of replicas in this and just use a size/
```
Comment (benesch) [inline src/catalog/src/builtin.rs:6748]: This seems fine to me, even for the long term. Probably not worth the comment unless you're planning to address imminently!

File: test/testdrive/system-cluster.td [hunk 4]
```diff
@@ -99,3 +104,12 @@ $ set-regex match=true|false replacement=<TRUE_OR_FALSE>
 
 > SHOW CLUSTER REPLICAS WHERE cluster = 'mz_system'
 mz_system r1 1 <TRUE_OR_FALSE>
+
+> SELECT COUNT(*) FROM mz_catalog.mz_cluster_replicas as r, mz_catalog.mz_clusters as c WHERE r.cluster_id = c.id and c.name = 'mz_support';
+0
+
+! ALTER CLUSTER mz_support SET (replication factor 1, size="2");
```
Comment (benesch) [inline test/testdrive/system-cluster.td:111]:
```suggestion
! ALTER CLUSTER mz_support SET (replication factor 1, size '2');
```

Double quotes are identifiers in SQL, and while we do support them in this position, it's a quirk of how options work, and using strings (single quotes) is more standard!

File: test/testdrive/system-cluster.td [hunk 5]
```diff
@@ -99,3 +104,12 @@ $ set-regex match=true|false replacement=<TRUE_OR_FALSE>
...
+contains: must be owner of CLUSTER mz_support
+
+$ postgres-execute connection=postgres://mz_support@${testdrive.materialize-internal-sql-addr}/materializeI
+ALTER CLUSTER mz_support SET (replication factor 0, size="1");
```
Comment (benesch) [inline test/testdrive/system-cluster.td:115]:
```suggestion
ALTER CLUSTER mz_support SET (replication factor 0, size '1');
```

(ditto)

File: src/catalog/src/builtin.rs (hunk 3 above)
Comment (jubrad) [reply src/catalog/src/builtin.rs:6748]: Yeah note to self I hadn't cleaned up.

File: src/adapter/src/catalog/open.rs [hunk 6]
```diff
@@ -1799,6 +1805,10 @@ impl BuiltinBootstrapClusterSizes {
...
```
Comment (benesch) [inline src/adapter/src/catalog/open.rs:1811]:
```suggestion
            Ok(self.support_cluster.clone())
```

**Review by benesch — COMMENTED** [review]
LGTM, but deferring approval to someone formally on the adapter team.

**Review by ParkMyCar — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**